htdocs/class/database/mysqldatabase.php

```diff
@@ -30,7 +30,7 @@
  * @package             class
  * @subpackage          database
  */
-class XoopsMySQLDatabase extends XoopsDatabase
+class XoopsMysqlDatabase extends XoopsDatabase
```

Comment:
Do you have an example that shows a real issue?

Unlike variable names, class names in PHP are not case sensitive. (PSR-0 and PSR-4 autoloaders use of class names are, but that doesn't apply here,) The factory method does go through some case manipulation, but it is not really necessary.

What is important is consistency. Since this is part of our public API, changing it means we will create inconsistencies in code from outside of core. I worry that these changes will create confusion and not solve anything.

Reply:
> The factory method does go through some case manipulation, but it is not really necessary.
This is right. So we should remove uneccessary code from the XoopsDatabaseFactory::getDatabaseConnection() method ... don't you think?

Reply:
I would accept a pull request that just removed the four occurrences of ucfirst() from XoopsDatabaseFactory. They do not serve any purpose.